### ISSUE TITLE
In notebook, kernel goes back to idle, but the line counter in the notebook doesn't update.

### ISSUE THREAD

**Issue by mccalluc** [hubmapconsortium/user_workspaces_server#66]
I'm running visualization notebooks generated by the code in
- https://github.com/hubmapconsortium/portal-ui/pull/2730

It works, but the `[*]` never updates to `[1]`, `[2]`, `[3]`, etc. Maybe the Jupiter UI just isn't getting some signal from the backend?
<img width="176" alt="Screen Shot 2022-07-22 at 3 31 42 PM" src="https://user-images.githubusercontent.com/730388/180515473-a17f0bae-aa9e-4d03-ab37-a6491ebea15d.png">

**Comment by jpuerto-psc** [comment]
Retrospective:

TLDR: Issue had to do with the websocket proxying that was built into the workspaces server.

The initial websocket proxying created two connections - one as a long-standing connection to the JupyterLab server to receive messages from that server and another one used to send messages from the front-end down to the JupyterLab server. This was causing messages to get lost, as JupyterLab was trying to send responses back to the connection that was proxying messages down to it.

Rather than having two connections, this was merged into one long-standing connection that is responsible for proxying both the messages that were being sent from the front-end down to Jupyter and the vice-versa.
